python/revng/scripting/local_daemon_profile.py

```diff
@@ -0,0 +1,73 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import os
+import shutil
+import subprocess
+from signal import SIGINT, SIGKILL
+from time import sleep
+from random import randint
+
+import requests
+
+from revng.scripting.projects import DaemonProject
+
+
+class LocalDaemonProcess():
+    def __init__(self, resume_dir: str = "", revng_executable: str = "") -> None:
+        self.port: int = randint(49152, 65535)
+        self.resume_dir: str = resume_dir
+        self.revng_executable: str = revng_executable
+        self.daemon_process: subprocess.Popen
+
+        self.start_daemon()
+
+    def get_client(self, connection_retries: int = 10) -> DaemonProject:
+        failed_retries = 0
+        while failed_retries <= connection_retries:
+            if self.is_server_running():
+                return DaemonProject(f"http://127.0.0.1:{self.port}")
+            failed_retries += 1
+            sleep(1)
+        raise RuntimeError(f"Couldn't connect to daemon server at http://127.0.0.1:{self.port}")
+
+    def start_daemon(self) -> None:
+        if not self.revng_executable:
+            self.revng_executable = "revng"
+            assert shutil.which(self.revng_executable) is not None
+
+        env = os.environ
+        if self.resume_dir:
+            env["REVNG_DATA_DIR"] = self.resume_dir
+
+        cli_args = [self.revng_executable, "daemon", "-b", f"tcp:127.0.0.1:{self.port}"]
+        self.daemon_process = subprocess.Popen(
+            cli_args, stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT, env=env
+        )
+
+    def stop_daemon(self) -> int:
+        self.daemon_process.send_signal(SIGINT)
+        try:
+            return self.daemon_process.wait(30.0)
+        except subprocess.TimeoutExpired:
+            self.daemon_process.send_signal(SIGKILL)
+
+        return self.daemon_process.wait()
+
+    def is_server_running(self) -> bool:
+        try:
+            requests.get(
+                f"http://127.0.0.1:{self.port}/status", timeout=5
+            )
+            return True
+        except requests.exceptions.ConnectionError:
+            return False
+
+    def __del__(self) -> None:
+        self.stop_daemon()
+
+
+class DaemonProfile():
```

Comment:
Both `DaemonProfile` and `CLIProfile` should take the `revng_executable as parameter in `__init__`